Assets/Scripts/UploadManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
+using System.Collections;
+using System.IO;
+using UnityEngine;
+
+using Firebase;
+using System.Threading.Tasks;
+using System.Threading;
+using Firebase.Storage;
+using Firebase.Extensions;
+using System;
+using UnityEngine.UI;
+using TMPro;
+
+namespace MoveToCode {
+    /// <summary>
+    /// UploadManager class uploads a CSV file to Firebase Storage
+    /// when user presses the Upload File button. It also
+    /// uses a progress bar to track the file upload process. 
+    /// </summary>
+    public class UploadManager : MonoBehaviour
+    {
+        private UploadBarController progressBar;
+        private TextMeshProUGUI progressText;
+        private long total_bytes = 0;
+        private long transferred_bytes = 0;
+        protected static string UriFileScheme = Uri.UriSchemeFile + "://";
+        private int count = 1;
+
+        // Initializes progress bar and progress text
+        private void Awake() {
+            progressText = GetComponentInChildren<Canvas>().GetComponentInChildren<TextMeshProUGUI>();
+            progressBar = GetComponentInChildren<Canvas>().GetComponentInChildren<Slider>().GetComponent<UploadBarController>();
+        }
+        
+        void Update() {
+            // Changes progress bar appropriately by bytes that have been uploaded (total value is out of 100)
+            if (total_bytes != 0)
+            {
+                progressBar.changeBytesUploaded((100 * transferred_bytes) / total_bytes);
+            }    
+        }
+
+        IEnumerator restartBtn() {
+            Debug.Log("Called!");
+            yield return new WaitForSeconds(5);
+            progressText.text = "Upload file";
+        }
+
+        // Gets the local filename as a URI relative to the persistent data path if the path isn't
+        // already a file URI.
+        protected virtual string PathToPersistentDataPathUriString(string filename) {
+            if (filename.StartsWith(UriFileScheme)) {
+                return filename;
+            }
+            return String.Format("{0}{1}/{2}", UriFileScheme, Application.persistentDataPath,
+                filename);
+        }
+    
+
+        // Gets the finished CSV file from LoggingManager and uploads
+        // it to Firebase 
+        public void UploadLog() {
```

Comment:
given this function has a couple "jobs", try extracting them out into different functions, the `FinishLogging(true)` is already done; now just do `LogToFireStore()` or something like that so the whole function will look like:
```C#
LoggingManager.instance.FinishLogging(true);
LogCSVToFireStore(); // first line should be UpdateProgressText("Started Upload");
```